ersilia/publish/inspect.py

```diff
@@ -0,0 +1,359 @@
+                        )
+                        check_passed = False
+
+                    source_url_works = (
+                        requests.head(file["Source Code"]).status_code != 404
+                    )
+                    if not source_url_works:
+                        details = (
+                            details
+                            + f"The URL ({file['Source Code']}) listed in the source code field of the metadata was not able to be accessed (Error code {requests.head(file['Source Code']).status_code}). Please verify that this link is accurate. "
+                        )
+                        check_passed = False
+
+                    s3_url_works = requests.head(file["S3"]).status_code != 404
+                    if not s3_url_works:
+                        details = (
+                            details
+                            + f"The URL ({file['S3']}) listed in the S3 field of the metadata was not able to be accessed (Error code {requests.head(file['S3']).status_code}). Please verify that this link is accurate. "
+                        )
+                        check_passed = False
+
+                    docker_url_works = (
+                        requests.head(file["DockerHub"]).status_code != 404
+                    )
+                    if not docker_url_works:
+                        details = (
+                            details
+                            + f"The URL ({file['DockerHub']}) listed in the DockerHub field of the metadata was not able to be accessed (Error code {requests.head(file['DockerHub']).status_code}). Please verify that this link is accurate. "
+                        )
+                        check_passed = False
+
+                else:
+                    details = (
+                        details
+                        + "A field in the metadata is empty. Please check that there are values listed for Publication, Source Code, S3, and DockerHub listed in the metadata. "
+                    )
+                    check_passed = False
+
+            except KeyError:  # If a given key not present in json file return false
+                check_passed = False
+                details = (
+                    details
+                    + "Not all required fields were found in the metadata. Please check that Publication, Source Code, S3, and DockerHub are all present in the metadata. "
+                )
+
+            except requests.exceptions.ConnectionError:
+                check_passed = False
+                details = (
+                    details
+                    + "Connection failed when trying to access a URL listed in the metadata. Please check that URLs are all accurate and accessible. "
+                )
+
+        if flag == 0:
+            return check_passed
+        if details:
+            return details
+        return "Check passed."
+
+    def folder_structure_complete(self, flag):
+        """
```

Comment:
Rename this method to `check_complete_folder_structure`

Reply:
done